File: Client/Captions/CaptionsLib/source/models/ICaptionsWriter.vb
'Project: ClipFlair (http://ClipFlair.codeplex.com)
'Filename: ICaptionsWriter.vb
'Version: 20121016

Imports System.Text
Imports System.IO
Imports Microsoft.SilverlightMediaFramework.Core.Accessibility.Captions

Namespace ClipFlair.CaptionsLib.Models

  Public Interface ICaptionsWriter

    Sub WriteCaptions(ByVal captions As CaptionRegion, ByVal path As String, ByVal theEncoding As Encoding)
    Sub WriteCaptions(ByVal captions As CaptionRegion, ByVal stream As Stream, ByVal theEncoding As Encoding)
    Sub WriteCaptions(ByVal captions As CaptionRegion, ByVal writer As TextWriter)

    Sub WriteHeader(ByVal writer As TextWriter)
    Sub WriteCaption(ByVal caption As CaptionElement, ByVal writer As TextWriter)
    Sub WriteFooter(ByVal writer As TextWriter)

  End Interface

End Namespace

File: Client/Captions/CaptionsLib/source/models/ICaptionsWriter.cs
//Project: ClipFlair (http://ClipFlair.codeplex.com)
//Filename: ICaptionsWriter.cs
//Version: 20131105

using System.IO;
using System.Text;

using Microsoft.SilverlightMediaFramework.Core.Accessibility.Captions;

namespace ClipFlair.CaptionsLib.Models
{

	public interface ICaptionsWriter
	{

		void WriteCaptions(CaptionRegion captions, string path, Encoding theEncoding);
		void WriteCaptions(CaptionRegion captions, Stream stream, Encoding theEncoding);

		void WriteCaptions(CaptionRegion captions, TextWriter writer);
		void WriteHeader(TextWriter writer);
		void WriteCaption(CaptionElement caption, TextWriter writer);

		void WriteFooter(TextWriter writer);
	}

}